src/group/organizationGroup/organizationGroup.controller.ts

```diff
@@ -240,6 +253,14 @@ export class OrganizationGroup {
     return res;
   }
 
+  static async searchGroups(query: Partial<GroupQuery>) {
+    const { hierarchy, name, ...filters } = query;
+    // the query makes sense only if any of these fields requested
+    if (!hierarchy && !name) return [];
+    return OrganizationGroup._organizationGroupTextSearch
+      .searchByNameAndHierarchy({ name, hierarchy }, filters);
```

Comment:
may call `searchByNameAndHierarchy` with object that have empty values (on existing keys)

Reply:
searchByNameAndHierarchy takes care of it